pkg/system/phase2_creating.go

```diff
@@ -353,3 +405,109 @@ func (r *Reconciler) SetDesiredAgentProfile(profileString string) string {
 	util.Panic(err)
 	return string(profileBytes)
 }
+
+// UpgradeSplitDB removes the old pvc and create a  new one with the same PV
+func (r *Reconciler) UpgradeSplitDB() error {
+	oldPvc := &corev1.PersistentVolumeClaim{
+		TypeMeta: metav1.TypeMeta{Kind: "PersistentVolumeClaim"},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "db-noobaa-core-0",
+			Namespace: options.Namespace,
+		},
+	}
+	if util.KubeCheck(oldPvc) {
+		r.Logger.Infof("UpgradeSplitDB: Old OVC found, upgrading...")
+		if err := r.UpgradeSplitDBSetReclaimPolicy(oldPvc, corev1.PersistentVolumeReclaimRetain); err != nil {
+			return err
+		}
+		if err := r.UpgradeSplitDBCreateNewPVC(oldPvc); err != nil {
+			return err
+		}
+		if err := r.UpgradeSplitDBSetReclaimPolicy(oldPvc, corev1.PersistentVolumeReclaimDelete); err != nil {
+			return err
+		}
+		if err := r.UpgradeSplitDBDeleteOldSTS(); err != nil {
+			return err
+		}
+		if err := r.UpgradeSplitDBDeleteOldPVC(oldPvc); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// UpgradeSplitDBSetReclaimPolicy sets the reclaim policy to reclaim parameter and checks it
+func (r *Reconciler) UpgradeSplitDBSetReclaimPolicy(oldPvc *corev1.PersistentVolumeClaim, reclaim corev1.PersistentVolumeReclaimPolicy) error {
+	pv := &corev1.PersistentVolume{
+		TypeMeta:   metav1.TypeMeta{Kind: "PersistentVolume"},
+		ObjectMeta: metav1.ObjectMeta{Name: oldPvc.Spec.VolumeName},
+	}
+	if !util.KubeCheck(pv) {
+		return fmt.Errorf("UpgradeSplitDBSetReclaimPolicy(%s): PV not found", reclaim)
+	}
+	if pv.Spec.PersistentVolumeReclaimPolicy != reclaim {
+		pv.Spec.PersistentVolumeReclaimPolicy = reclaim
+		if pv.Spec.ClaimRef != nil &&
+			pv.Spec.ClaimRef.Name == oldPvc.Name &&
+			pv.Spec.ClaimRef.Namespace == oldPvc.Namespace {
+			pv.Spec.ClaimRef = nil
+		}
+		util.KubeUpdate(pv)
+		if !util.KubeCheck(pv) {
+			return fmt.Errorf("UpgradeSplitDBSetReclaimPolicy(%s): PV not found after update", reclaim)
+		}
+		if pv.Spec.PersistentVolumeReclaimPolicy != reclaim {
+			return fmt.Errorf("UpgradeSplitDBSetReclaimPolicy(%s): PV reclaim policy could not be updated", reclaim)
+		}
+	}
+	return nil
+}
+
+// UpgradeSplitDBCreateNewPVC creates new pvc and checks it
+func (r *Reconciler) UpgradeSplitDBCreateNewPVC(oldPvc *corev1.PersistentVolumeClaim) error {
+	newPvc := &corev1.PersistentVolumeClaim{
+		TypeMeta: metav1.TypeMeta{Kind: "PersistentVolumeClaim"},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "db-" + r.NooBaaDB.Name + "-0",
+			Namespace: options.Namespace,
+		},
+		Spec: oldPvc.Spec,
+	}
+	util.KubeCreateSkipExisting(newPvc)
+	if !util.KubeCheck(newPvc) {
```

Comment:
Do you think we should add a short sleep here to avoid a restart of reconcile in most cases?